unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/VoiceChatHUD/VoiceChatWindowController.cs

```diff
@@ -0,0 +1,436 @@
+using DCL;
+using DCL.Interface;
+using DCL.SettingsCommon;
+using SocialFeaturesAnalytics;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using static DCL.SettingsCommon.GeneralSettings;
+
+public class VoiceChatWindowController : IHUD
+{
+    const float MUTE_STATUS_UPDATE_INTERVAL = 0.3f;
+    private const string TALKING_MESSAGE_YOU = "You";
+    private const string TALKING_MESSAGE_JUST_YOU_IN_THE_VOICE_CHAT = "Just you in the voice chat";
+    private const string TALKING_MESSAGE_NOBODY_TALKING = "Nobody is talking";
+    private const string TALKING_MESSAGE_SEVERAL_PEOPLE_TALKING = "Several people talking";
+
+    public IVoiceChatWindowComponentView VoiceChatWindowView => voiceChatWindowView;
+    public IVoiceChatBarComponentView VoiceChatBarView => voiceChatBarView;
+
+    private UserProfile ownProfile => userProfileBridge.GetOwn();
+
+    private IVoiceChatWindowComponentView voiceChatWindowView;
+    private IVoiceChatBarComponentView voiceChatBarView;
+    private IUserProfileBridge userProfileBridge;
+    private IFriendsController friendsController;
+    private ISocialAnalytics socialAnalytics;
+    private DataStore dataStore;
+    private Settings settings;
+    private readonly HashSet<string> trackedUsersHashSet = new HashSet<string>();
+    private readonly Queue<VoiceChatPlayerComponentView> playersPool;
+    private readonly Dictionary<string, VoiceChatPlayerComponentView> currentPlayers;
```

Comment:
i think the controller should not cache views, instead it should save the models